### PR TITLE
refactor: refactor api interface

### PR DESCRIPTION
**Pull request by jemmyshin**
refactor the api interface to follow openai style

### CODE REVIEW

File: open_gpt/models/generation.py [hunk 1]
```diff
@@ -313,7 +324,7 @@ def generate(self, prompt: str, max_context_length: int = CONTEXT_LEN, **kwargs)
         skip_special_tokens = kwargs.pop("skip_special_tokens", True)
         echo = kwargs.pop("echo", False)
 
-        max_length = kwargs.pop("max_length", max_context_length)
+        max_length = MAX_LENGTH
```
Comment (numb3r3) [inline open_gpt/models/generation.py:327]:
```suggestion
        max_length = kwargs.get('max_length', MAX_LENGTH)
```

File: open_gpt/models/generation.py [hunk 2]
```diff
@@ -336,4 +347,17 @@ def generate(self, prompt: str, max_context_length: int = CONTEXT_LEN, **kwargs)
                 skip_special_tokens=skip_special_tokens,
             )
 
-            return text
+            resp = {
+                "choices": [
+                    {"index": 0, "text": text, "finish_reason": "length"},
```
Comment (numb3r3) [inline open_gpt/models/generation.py:352]: shoud't the `finish_reason = 'stop'`

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: open_gpt/models/generation.py [hunk 3]
```diff
@@ -336,4 +347,17 @@ def generate(self, prompt: str, max_context_length: int = CONTEXT_LEN, **kwargs)
...
+                ],
+                "prompt": prompt,
+                "usage": {
+                    "prompt_tokens": 0,
```
[the remark this answers is not in the document]
Comment (jemmyshin) [reply open_gpt/models/generation.py:367]: in openai, `prompt_tokens` means the length of the input prompt. However, they don't consider the case where `past_key_values` exist. So in our scenario, `prompt_tokens` is the length of `past_key_values`, and `input_length` is the length of the input prompt.

File: open_gpt/models/generation.py (hunk 2 above)
Comment (jemmyshin) [reply open_gpt/models/generation.py:363]: `stop` only happens when we set the `stop_str` and it appears in generated text

**Review by numb3r3 — APPROVED** [review]
LGTM

**Pull request merged**